pkg/tools/cache/expiration_cache.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+)
+
+const (
+	DefaultExpiration = 2 * time.Minute
+	DefaultGCInterval = time.Minute
+)
+
+type item struct {
+	object         interface{}
+	expirationTime time.Time
+}
+
+type Cache struct {
```

Comment:
looks like this cache is not needed yet. maybe we could implement it in another PR.